testing/tests/DevExpress.ui.widgets/listParts/commonTests.js

```diff
@@ -1788,6 +1789,77 @@ QUnit.module('selection', moduleSetup, () => {
     });
 });
 
+QUnit.module('selectByClick option', ()=> {
+    QUnit.test('selectByClick option should be changed after initialization', function(assert) {
+        const list = $('#list').dxList({ items: [1, 2] }).dxList('instance');
+
+        list.option('selectByClick', false);
+
+        const isSelectByClick = list.option('selectByClick');
+
+        assert.strictEqual(isSelectByClick, false);
+    });
+
+    QUnit.test('onSelectionChanged event shouldn`t be called on item click if selectByClick is false', function(assert) {
+        const onSelectionChangedSpy = sinon.spy();
+        const $list = $('#list').dxList({
+            items: [1, 2],
+            showSelectionControls: true,
+            selectByClick: false,
+            selectionMode: 'multiple',
+            onSelectionChanged: onSelectionChangedSpy,
+        });
+        const $items = $list.find(`.${LIST_ITEM_CLASS}`);
+
+        $items.eq(0).trigger('dxclick');
+        $items.eq(1).trigger('dxclick');
+
+        assert.strictEqual(onSelectionChangedSpy.callCount, 0);
+    });
+
+    QUnit.test('onSelectionChanged event should call on item click if selectByClick is true', function(assert) {
+        const onSelectionChangedSpy = sinon.spy();
+        const $list = $('#list').dxList({
+            items: [1, 2],
+            showSelectionControls: true,
+            selectByClick: true,
+            selectionMode: 'multiple',
+            onSelectionChanged: onSelectionChangedSpy,
+        });
+
+        const $items = $list.find(`.${LIST_ITEM_CLASS}`);
+
+        $items.eq(0).trigger('dxclick');
+        assert.strictEqual($items.eq(0).hasClass(LIST_ITEM_SELECTED_CLASS), true, 'item has selected class');
+
+        $items.eq(1).trigger('dxclick');
+        assert.strictEqual($items.eq(1).hasClass(LIST_ITEM_SELECTED_CLASS), true, 'item has selected class');
+
+        assert.strictEqual(onSelectionChangedSpy.callCount, 2);
+    });
+
+    QUnit.test('onSelectionChanged event shouldn`t be called on item click if selectByClick=true, selectionMode=single', function(assert) {
+        const onSelectionChangedSpy = sinon.spy();
+        const $list = $('#list').dxList({
+            items: [1, 2],
+            showSelectionControls: true,
+            selectByClick: true,
+            selectionMode: 'single',
+            onSelectionChanged: onSelectionChangedSpy,
+        });
+
+        const $items = $list.find(`.${LIST_ITEM_CLASS}`);
+
+        $items.eq(0).trigger('dxclick');
+        assert.strictEqual($items.eq(0).hasClass(LIST_ITEM_SELECTED_CLASS), true, 'item has selected class');
+
+        $items.eq(1).trigger('dxclick');
+        assert.strictEqual($items.eq(1).hasClass(LIST_ITEM_SELECTED_CLASS), true, 'item has selected class');
```

Comment:
Let's check that $items.eq(0) doesn't has selected class after click on a new one